apps/builder/app/builder/features/style-panel/sections/borders/border-style.tsx

```diff
@@ -1,28 +1,103 @@
 import type { StyleProperty } from "@webstudio-is/css-data";
+import { toValue } from "@webstudio-is/css-engine";
+import {
+  Grid,
+  theme,
+  ToggleGroup,
+  ToggleGroupItem,
+  Tooltip,
+} from "@webstudio-is/design-system";
+import {
+  DashBorderIcon,
+  DashedBorderIcon,
+  DottedBorderIcon,
+  SmallXIcon,
+} from "@webstudio-is/icons";
+import { toPascalCase } from "../../shared/keyword-utils";
+import { PropertyName } from "../../shared/property-name";
 import type { RenderCategoryProps } from "../../style-sections";
-import { BorderProperty } from "./border-property";
+import { deleteAllProperties, setAllProperties } from "./border-utils";
 
-const borderPropertyOptions = {
-  borderTopStyle: {},
-  borderRightStyle: {},
-  borderLeftStyle: {},
-  borderBottomStyle: {},
-} as const satisfies Partial<{ [property in StyleProperty]: unknown }>;
+const borderStyleProperties = [
+  "borderTopStyle",
+  "borderRightStyle",
+  "borderLeftStyle",
+  "borderBottomStyle",
+] as const satisfies readonly StyleProperty[];
+
+const borderStyleValues = [
+  { value: "none", icon: SmallXIcon },
```

Comment:
icon here should start capitalize if you are passing a component, alternative is to pass an instance: 

icon: <SmallXIcon />
or
Icon: SmallXIcon